code/scripting/api/objs/object.cpp

```diff
@@ -202,6 +202,24 @@ ADE_VIRTVAR(HitpointsLeft, l_Object, "number", "Hitpoints an object has left", "
 	return ade_set_args(L, "f", objh->objp->hull_strength);
 }
 
+ADE_VIRTVAR(SimHitpointsLeft, l_Object, "number", "Simulated hitpoints an object has left", "number", "Simulated hitpoints left, or 0 if handle is invalid")
+{
+	object_h *objh = NULL;
+	float f = -1.0f;
+	if (!ade_get_args(L, "o|f", l_Object.GetPtr(&objh), &f))
+		return ade_set_error(L, "f", 0.0f);
+
+	if (!objh->IsValid())
+		return ade_set_error(L, "f", 0.0f);
+
+	//Set sim hull strength.
+	if (ADE_SETTING_VAR) {
+		objh->objp->sim_hull_strength = f;
```

Comment:
Is a negative value valid? If not then a range check and error might be a good idea here.

Reply:
Good call.

Reply:
Looks like negative values are, in fact, valid.  For regular hull, they'll just cause the ship to explode immediately.  For sim hull, the value is range-checked when the sexp is called.